commitlint.config.ts

```diff
@@ -20,6 +20,32 @@ function isBigBlock(line: string) {
     return (line.length == bigBlockDelimiter.length) && (line.indexOf("```") == 0);
 }
 
+function isUpperCase(letter: string) {
```

Comment:
Before I merge this PR I want to raise awareness of this nitpick that might become a potential problem in the future, if some developer makes use of this function without being careful: see how the parameter is called "letter", but the type is "string" instead of "char". Now, it turns out the type `char` doesn't exist on TypeScript so we will need to protect ourselves from this pitfall at runtime instead of at compile time (better than nothing...). So let's do some checks before performing the logic in the function, as in "preconditions". For this function I guess the only precondition is that letter's length is just one character (if it's not, just throw an exception). After you're done adding this precondition, add others to other functions, such as isLowerCase, mightBeUrl (see the param is called "line" in that case, so that's also an assumption), and the other two

Reply:
Done